presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -690,6 +740,7 @@ public String toString()
         private final Type type;
         private final Optional<Symbol> hashSymbol;
         private final Optional<List<Symbol>> keys;
+        private final Optional<NullPartitioningPolicy> nullPartitioningPolicy;
 
         public static PartitioningProperties arbitrary()
         {
```

Comment:
I have realized that my change breaks this function. I'll submit a fix for this when I resolve your comments.